docs/libcudacxx/extended_api/numeric/add_overflow.rst

```diff
@@ -0,0 +1,92 @@
+.. _libcudacxx-extended-api-numeric-add_overflow:
+
+``cuda::add_overflow``
+==========================
+
+.. code:: cpp
+
+   template <class T>
+   struct overflow_result;
+
+   template <class Result = /*unspecified*/, class Lhs, class Rhs>
+   [[nodiscard]] __host__ __device__ inline constexpr
+   overflow_result</*see-below*/> add_overflow(Lhs lhs, Rhs rhs) noexcept; // (1)
+
+   template <class Result, class Lhs, class Rhs>
+   [[nodiscard]] __host__ __device__ inline constexpr
+   bool add_overflow(Result& result, Lhs lhs, Rhs rhs) noexcept; // (2)
+
+The function ``cuda::add_overflow`` performs addition of two values ``lhs`` and ``rhs`` with overflow checking. The result is the same as if the operands were first promoted to an infinite precision signed type, added together and the result truncated to the type of the return value.
+
+**Parameters**
+
+- ``result``: The result of the addition (2).
+- ``lhs``: The left-hand side operand (1, 2).
+- ``rhs``: The right-hand side operand (1, 2).
+
+**Return value**
+
+- (1) Returns an :ref:`overflow_result <libcudacxx-extended-api-numeric-overflow_result>` object that contains the result of the addition and a boolean indicating whether an overflow occurred. If the ``Result`` type is specified, it will be used as the type of the result, otherwise the common type of ``Lhs`` and ``Rhs`` is used.
+- (2) Returns a boolean indicating whether an overflow occurred, and if no overflow occurred,
+  the result is stored in the provided ``result`` variable.
+
+**Constraints**
+
+- ``Result``, ``Lhs``, and ``Rhs`` must be cv-unqualified integer types.
+
+**Performance considerations**
+
+- TODO
+
+Example
+-------
+
+.. code:: cuda
+
+    #include <cuda/numeric>
+    #include <cuda/std/cassert>
+    #include <cuda/std/limits>
+
+    __global__ void kernel()
+    {
+        constexpr auto int_max = cuda::std::numeric_limits<int>::max();
+        constexpr auto int_min = cuda::std::numeric_limits<int>::min();
+
+        // cuda::add_overflow(lhs, rhs) returning common type of lhs and rhs
+        if (auto result = cuda::add_overflow(1, int_max))
+        {
+            assert(result.value == int_min);
+        }
+        else
+        {
+            assert(false); // should not be reached
+        }
+
+        // cuda::add_overflow<Result>(lhs, rhs) with explicit return type
+        if (auto result = cuda::add_overflow<long long>(-1, int_min))
```

Comment:
I would prefer if this would be the exact same example as above, and a comment that highlights that enlarging the type can avoid overflow